app/Actions/Album/Tracks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\Models\Album;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+
+class Tracks extends Action
+{
+	/**
+	 * @param string $albumID the ID of the album
+	 * @param mixed  $value   the value to be set
+	 */
+	public function set(string $albumID, UploadedFile $value): void
+	{
+		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
+		if ($album->track_id != null) {
+			Storage::delete("tracks/$album->track_id.xml");
+		}
+
+		$new_track_id = uniqid();
```

Comment:
`uniqid` is time-based and hence not really secure. If you have uploaded an album at a certain point of time, an attacker might guess the ID of the track and download it subsequently, because an attacker only needs to scan a small range of values. This could infringe privacy.

We just got rid of time-based IDs and use 24 alpha-numeric characters for such things. Also a (cryptographically secure hash) over the track would be fine.